### ISSUE TITLE
Feature request: Gapless play

### ISSUE THREAD

**Issue by Ulrichgau** [owntone/owntone-server#1272]
Hello, is it possible to implement a gapless play function? I would like to hear saved songs without the gap. Also for zapping through radio stations, it would be nice if hte next two streams would be already opened in order to reduce the gaps.

Best regards
Uli

**Comment by ejurgensen** [comment]
Last time I looked at gapless my conclusion was that the gaps come from ffmpeg's decoding of the files, and I could not figure out a way to avoid the gaps it produced. I'm sure there is one, but I admit I gave up on it.

Zapping radio stations is an interesting idea, but to be honest I don't think I will ever get around to doing that.

**Comment by ubik3000** [comment]
Hello,
I have played mixed compilations (so gapless tracks) (in mp3 format) in Owntone to Airplay Receiver (Shairport4W) and to Chromecast Audio and the playback was gapless. I am not sure what @Ulrichgau means by saved songs...

**Comment by braunsch** [comment]
I second the enhancement request: A lot of the music I am listening to is either classical, or live performances - where it is irritating to have the applause between songs interrupted by a short “tick” because a new track starts. 
I always thought you’d have to buffer the stream for a very short time to enable gapless (giving a program like ffmpeg time to start working on the next song) - but I’m not the genius programmer here.

**Comment by ejurgensen** [comment]
Yes, I would also love to support this. OwnTone already reads ahead by quite a bit, so the gap isn't related to timing. As I recall, it comes from the decoding, so if you look at it as a PCM stream it looks a bit like this (AA being bytes with audio, 00 is the gap):

```
AA AA AA AA AA AA 00 00 00 00 00 00 (new track starts) AA AA AA AA
```

I never found out why ffmpeg was padding with silence, and if there was a way to avoid it.

**Comment by braunsch** [comment]
over a year later… 
I found a conversation that points to a problem with FFmpeg at the core of the inability to support truly gapless playback (see here: https://trac.ffmpeg.org/ticket/9140). Many programs seem to have the same problem when using FFmpeg for transcoding/decoding (tested by me: Emby, Foobar2000, BubbleUPnP w/ PCM decoding, mPlayer). Others, who seem not to use FFmpeg seem to not have the problem (jRiver Media Center, anything Apple). 

I am not sure if there is a way to make those gaps go away by putting different or additional parameters to FFmpeg?
